### DISCUSSION TITLE
Edit document source (raw Markdown)

### DISCUSSION THREAD

**Discussion by filimonov** [outline/outline#1570] [General] [answered]
Your WYSIWYG is fast and nice... But I hate WYSIWYG, i've tried outline because it's said 'markdown compatible'. 

But once i start copying and pasting some texts - i' immediately hit typical  WYSIWYG limitation. 

For example if i start a document with a piece of code (with 3 backticks) there is no simple way to put something before that piece of code (all attempt to type something 'before' lead nowhere). 

Can you give just an 'edit source' button?

**Comment by tommoor** [comment, marked as answer]
Hi @filimonov – we've considered a raw mode several times, it might still happen but such a thing is incompatible with a lot of the more advanced features that we hope to build – so enabling it would come at quite a cost.

I agree that some of these edge cases in the editor could still be improved, for your specific example – you can always put the caret in the title field and hit Enter to get a newline at the top of the document.
